### PR TITLE
detect inout arguments when lhs is a Variable

### PR DESCRIPTION
**Pull request by saidctb**

### CODE REVIEW

File: tests/epyccel/test_epyccel_functions.py [hunk 1]
```diff
@@ -154,6 +154,35 @@ def f8(x,b):
...
+        x += 1
+
+    f = epyccel(f9)
+
+    x = np.zeros(10, dtype='int64')
+    x_expected = x.copy()
+
+    f9(x)
+    f(x_expected)
+    assert np.allclose(x, x_expected, rtol=1e-15, atol=1e-15)
```
Comment (yguclu) [inline tests/epyccel/test_epyccel_functions.py:170]: Shouldn't we check for exact equality here? This is an integer array...

File: tests/epyccel/test_epyccel_functions.py [hunk 2]
```diff
@@ -154,6 +154,35 @@ def f8(x,b):
...
+        x[:] += 1
+
+    f = epyccel(f10)
+
+    x = np.zeros(10, dtype='int64')
+    x_expected = x.copy()
+
+    f10(x)
+    f(x_expected)
+    assert np.allclose(x, x_expected, rtol=1e-15, atol=1e-15)
```
Comment (yguclu) [inline tests/epyccel/test_epyccel_functions.py:184]: Same comment as above.

File: tests/epyccel/test_epyccel_functions.py (hunk 1 above)
Comment (saidctb) [reply tests/epyccel/test_epyccel_functions.py]: done

File: tests/epyccel/test_epyccel_functions.py (hunk 2 above)
Comment (saidctb) [reply tests/epyccel/test_epyccel_functions.py]: done

**Pull request merged**